sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryAvroUtils.java

```diff
@@ -205,6 +209,9 @@ private static Object convertRequiredField(
         byte[] bytes = new byte[byteBuffer.limit()];
         byteBuffer.get(bytes);
         return BaseEncoding.base64().encode(bytes);
+      case "DATE":
+        verify(v instanceof Utf8, "Expected Utf8, got %s", v.getClass());
```

Comment:
why is this not `CharSequence` (as done for strings)?

If it is expected to be, then this should be a shared case